packages/cozy-harvest-lib/src/helpers/accounts.js

```diff
@@ -90,11 +91,21 @@ export const mergeAuth = (account, authData) => ({
  * @return {object}           io.cozy.accounts attributes
  */
 export const updateTwoFaCode = (account, code) => ({
-  ...account,
-  // reset the state,
-  state: null,
+  // reset the state since the konnector is listening it
+  ...resetState(account),
   twofa_code: code
 })
+
+/**
+ * Reset the account state
+ * @param {[type]} account Account document
+ * @return {object}        Changed account document
+ */
+export const resetState = account => ({
+  ...account,
+  state: null
+})
```

Comment:
This code could be moved to cozy-libs/doctypes.

Reply:
We agree, it's what we aim to do as soon as possible, as we speak a few weeks ago.
For now we need to ship quickly, I suggest to keep thoses models/helpers in Harvest and extract them when everything will work.